codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,253 @@
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadSimpleString.byteBufCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(RedisMessageType.SIMPLE_STRING,
+                                                                content.toString(CharsetUtil.UTF_8));
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#ERROR}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createError(String content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadErrors.stringCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(RedisMessageType.ERROR, content);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} from {@code content} using {@link RedisMessageType#ERROR}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createError(ByteBuf content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadErrors.byteBufCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(RedisMessageType.ERROR,
+                                                                content.toString(CharsetUtil.UTF_8));
+    }
+
+    /**
+     * Creates a {@link IntegerRedisMessage} for the given {@code content}.
+     *
+     * @param value the message content.
+     * @return a new {@link IntegerRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createInteger(long value) {
+        return new IntegerRedisMessage(value);
+    }
+
+    /**
+     * Creates a {@link BulkStringRedisMessage} with {@code null} content.
+     *
+     * @return a {@link BulkStringRedisMessage} with {@code null} content.
+     */
+    public static RedisMessage nullBulkString() {
+        return NULL_BULK_STRING;
+    }
+
+    /**
+     * Creates a {@link BulkStringRedisMessage} with empty content.
+     *
+     * @return a {@link BulkStringRedisMessage} with empty content.
+     */
+    public static RedisMessage emptyBulkString() {
+        return EMPTY_BULK_STRING;
+    }
+
+    /**
+     * Creates a {@link BulkStringRedisMessage} for the given {@code content}.
+     * Note that non-empty {@code content} results in {@link io.netty.util.ReferenceCounted} instances.
+     *
+     * @param content the content, must not be {@code null}.
+     * @return a {@link BulkStringRedisMessage} for the given {@code content}
+     */
+    public static RedisMessage createBulkString(ByteBuf content) {
+        ObjectUtil.checkNotNull(content, "content");
```

Comment:
should we just remove `nullBulkString()` and `emptyBulkString()` and change this method to:

```java
if (content == null) {
  return EMPTY_BULK_STRING;
}
if (content.isReadable()) {
  return EMPTY_BULK_STRING;
}
return new RefCountedBulkStringRedisMessage(content);
```